Mathlib/Algebra/Order/WithTop/Untop0.lean

```diff
@@ -0,0 +1,90 @@
+/-
+Copyright (c) 2025 Stefan Kebekus. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Stefan Kebekus
+-/
+
+import Mathlib.Algebra.Order.AddGroupWithTop
+import Mathlib.Algebra.Order.Ring.WithTop
+
+/-!
+# Conversion from WithTop to Base Type
+
+For types α that are instances of `Zero`, we provide a convenient conversion,
+`WithTop.untop₀`, that maps elements `a : WithTop α` to `α`, by mapping `⊤` to
+zero.
+
+For settings where `α` has additional structure, we provide a large number of
+simplifier lemmas, akin to those that already exists for `ENat.toNat`.
+-/
+
+namespace WithTop
+variable {α : Type*}
+
+section Zero
+variable [Zero α]
+
+/-- Conversion from `WithTop α` to `α`, mapping `⊤` to zero. -/
+def untop₀ [Zero α] (a : WithTop α) : α := a.untopD 0
+
+/-!
+## Simplifying Lemmas in cases where α is an Instance of Zero
+-/
+
+@[simp]
+lemma untop₀_eq_zero {a : WithTop α} :
+    a.untop₀ = 0 ↔ a = 0 ∨ a = ⊤ := by simp [untop₀]
+
+@[simp]
+lemma untop₀_top : untop₀ ⊤ = (0 : α) := by simp [untop₀]
+
+@[simp]
+lemma untop₀_zero : untop₀ 0 = (0 : α) := by simp [untop₀]
+
+@[simp]
+lemma untop₀_coe (a : α) : (a : WithTop α).untop₀ = a := rfl
+
+end Zero
+
+/-!
+## Simplifying Lemmas in cases where α is an AddMonoid
+-/
+@[simp]
+lemma untop₀_add [AddMonoid α] {a b : WithTop α} (ha : a ≠ ⊤) (hb : b ≠ ⊤) :
```

Comment:
This is true for any `untopD`, not only `untop₀`.

Reply:
Generalized lemma and stated existing lemma as special case.